op-service/txmgr/txmgr.go

```diff
@@ -297,19 +294,26 @@ func (m *SimpleTxManager) sendTx(ctx context.Context, tx *types.Transaction) (*t
 
 	sendState := NewSendState(m.cfg.SafeAbortNonceTooLowCount, m.cfg.TxNotInMempoolTimeout)
 	receiptChan := make(chan *types.Receipt, 1)
-	sendTxAsync := func(tx *types.Transaction) {
-		defer wg.Done()
-		m.publishAndWaitForTx(ctx, tx, sendState, receiptChan)
+	publishAndWait := func(tx *types.Transaction, bumpFees bool) *types.Transaction {
+		tx, published := m.publishTx(ctx, tx, sendState, bumpFees)
```

Comment:
Should publishTx be called synchronously?  It shouldn't block for long, and as long as it's blocking we probably don't want to be retrying it.

Reply:
Hmm, I guess if it does error out though you are relying on the below loop timeout to handle the retry.  I wonder if retry for publish error should be a different timeout than retry-for-stuck-in-txpool?

Reply:
the reason it's called synchronously is because it modifies the `tx`... if it were async, we'd have to build messaging between the two go routines to get the updated tx back

Reply:
right sorry, I was suggesting it should be called synchronously, somehow having missed it is already being called that way.